src/sqlast/value.rs

```diff
@@ -21,6 +21,12 @@ pub enum Value {
     Time(String),
     /// Timestamp literals, which include both a date and time
     Timestamp(String),
+    /// Time intervals
+    Interval {
+        value: String,
+        start_qualifier: SQLIntervalQualifier,
+        end_qualifier: SQLIntervalQualifier,
```

Comment:
Given what I learned about INTERVALs in SQL (see my comment in #62), it was counter-intuitive to me that this stores `HOUR(2)` as if it was equivalent to `HOUR(2) TO HOUR(2)`.

I think it makes more sense to model this as `leading_field: SQLDateTimeField, leading_precision: Option<u64>, last_field: Option<SQLDateTimeField>, fractional_seconds_precision: Option<u64>`. (If we want to support Postgres' `INTERVAL '2 years 13 months'`, the `leading_field` will have to be an Option too.)

* `SECONDS(p,q)` would parse to `{leading_field: Seconds, leading_precision: p, last_field: None, fractional_seconds_precision: q}`
* `A(p) TO B(q)` would parse to `{leading_field: A, leading_precision: p, last_field: B, fractional_seconds_precision: q}`. We might want to forbid `A = Seconds`, and `q = Some` when `B != Seconds`.
* `A(p)` would parse to `{leading_field: A, leading_precision: p, last_field: None, fractional_seconds_precision: None}`

Reply:
> I think it makes more sense to model this as...

Sure, that works for me. Updated!

> If we want to support Postgres' `INTERVAL '2 years 13 months'`, the leading_field will have to be an Option too.

I don't currently have a need to support Postgres's nonstandard interval syntax, so I'm going to omit for now. I've left a note, though.